File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        if(head.next == null) return null;
        ListNode first = head;
        ListNode second = head;
        ListNode temp = null;
        
        while( first!=null && first.next!=null){
            temp=second;
            first=first.next.next;
            second=second.next;
        }
        temp.next=temp.next.next;
        return head;
    }
}

    

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head.next:
            return None
        f,s=head,head
        temp=None
        while f and f.next:
            temp=s
            f=f.next.next
            s=s.next
        temp.next=temp.next.next
        return head
    